lib/ddtrace/contrib/presto/instrumentation.rb

```diff
@@ -0,0 +1,83 @@
+require 'ddtrace/pin'
+require 'ddtrace/ext/net'
+require 'ddtrace/ext/sql'
+require 'ddtrace/ext/app_types'
+require 'ddtrace/contrib/presto/ext'
+
+module Datadog
+  module Contrib
+    module Presto
+      # Instrumentation for Presto integration
+      module Instrumentation
+        # Instrumentation for Presto::Client::Client
+        module Client
+          def self.included(base)
+            base.send(:prepend, InstanceMethods)
+          end
+
+          # Instance methods for Presto::Client
+          module InstanceMethods
+            def run(query)
+              datadog_pin.tracer.trace(Ext::SPAN_QUERY) do |span|
+                decorate!(span)
+                span.resource = query
+                span.span_type = Datadog::Ext::SQL::TYPE
+                span.set_tag(Ext::TAG_QUERY_ASYNC, false)
+                super(query)
+              end
+            end
+
+            def query(query, &blk)
+              datadog_pin.tracer.trace(Ext::SPAN_QUERY) do |span|
+                decorate!(span)
+                span.resource = query
+                span.span_type = Datadog::Ext::SQL::TYPE
+                span.set_tag(Ext::TAG_QUERY_ASYNC, !blk.nil?)
+                super(query, &blk)
+              end
+            end
+
+            def kill(query_id)
+              datadog_pin.tracer.trace(Ext::SPAN_KILL) do |span|
+                decorate!(span)
+                span.resource = Ext::SPAN_KILL
+                span.span_type = datadog_pin.app_type
+                # ^ not an SQL type span, since there's no SQL query
+                span.set_tag(Ext::TAG_QUERY_ID, query_id)
+                super(query_id)
+              end
+            end
+
+            private
+
+            def datadog_pin
+              @datadog_pin ||= Datadog::Pin.new(
+                Datadog.configuration[:presto][:service_name],
+                app: Ext::APP,
+                app_type: Datadog::Ext::AppTypes::DB,
+                tracer: Datadog.configuration[:presto][:tracer]
+              )
+            end
+
+            def decorate!(span)
+              span.service = datadog_pin.service
```

Comment:
You'll want to implement analytics sample rate too, like so: https://github.com/DataDog/dd-trace-rb/blob/73e841baa4fe0a490cff53f9dae54069dd6501a4/lib/ddtrace/contrib/active_record/events/sql.rb#L42-L45